### PR TITLE
Bugfix/restriction filtering

### PR DESCRIPTION
**Pull request by halfhorst**
A few more bugfixes

### CODE REVIEW

File: src/vivarium_inputs/utilities.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def normalize_sex(data: pd.DataFrame, fill_value) -> pd.DataFrame:
 
 def normalize_year(data: pd.DataFrame) -> pd.DataFrame:
     binned_years = gbd.get_estimation_years()
-    years = {'annual': list(range(binned_years.min(), binned_years.max() + 1)), 'binned': binned_years}
+    years = {'annual': list(range(min(binned_years), max(binned_years) + 1)), 'binned': binned_years}
```
Comment (kfwilson) [inline src/vivarium_inputs/utilities.py]: why is this the right min/max to use? I didn't realize they would give different results

File: src/vivarium_inputs/utilities.py [hunk 2]
```diff
@@ -294,7 +294,7 @@ def filter_data_by_restrictions(data: pd.DataFrame, entity: Union[RiskFactor, Ca
     elif not restrictions.male_only and restrictions.female_only:
         sexes = gbd.FEMALE
     else:  # not male only and not female only
-        sexes = gbd.FEMALE + gbd.MALE
+        sexes = gbd.FEMALE + gbd.MALE + gbd.COMBINED
```
Comment (kfwilson) [inline src/vivarium_inputs/utilities.py]: hmm wait actually where are we using this that we have sex_id = 3 and why do we need to keep those rows?

File: src/vivarium_inputs/utilities.py (hunk 1 above)
Comment (michaella21) [reply src/vivarium_inputs/utilities.py]: I included in current my working branch also :) yeah we can't use the .min() for the list :(

File: src/vivarium_inputs/utilities.py (hunk 2 above)
Comment (michaella21) [reply src/vivarium_inputs/utilities.py]: I think it's for the case when data frame only has sex id 3?

File: src/vivarium_inputs/utilities.py (hunk 1 above)
Comment (halfhorst) [reply src/vivarium_inputs/utilities.py]: it is a plain python list, not a series or dataframe

Comment (kfwilson) [reply src/vivarium_inputs/utilities.py]: ohh didn't realize get_estimation_years gave back a list

File: src/vivarium_inputs/utilities.py (hunk 2 above)
Comment (halfhorst) [reply src/vivarium_inputs/utilities.py]: this filtering happens before normalization, so sex_id 3 still can exist in the data

Comment (halfhorst) [reply src/vivarium_inputs/utilities.py]: it led to an empty dataframe. Incidentally, maybe we should catch that -- if you filter everything away, raise some sort of exception

**Pull request merged**